### PR TITLE
feat: events notifications with Redis pub/sub

### PR DESCRIPTION
**Pull request by agusaldasoro**
1. Adds Redis as a Pub/Sub system to communicate between nodes of the Social Server, the nodes will communicate through one channel all event updates that will trigger the notification system. The service already uses Redis for the users' cache, but another instance of Redis will be initiated for the notifications system.

As it's a Redis limitation when a client is subscribed to the same channel that it's publishing, then their own messages will be retrieved. That's why the notifications between peers in the same node will go trough Redis anyway.

2. Adds the implementation of the RPC message:
  ```proto
  rpc SubscribeFriendshipEventsUpdates(Payload)
        returns (stream SubscribeFriendshipEventsUpdatesResponse) {}
  ```
3. Created an internal proto file for the messages shared in the pub/sub system

Fixes: https://github.com/decentraland/dservices/issues/63
Fixes: https://github.com/decentraland/dservices/issues/64

### CODE REVIEW

**Review by Julieta11 — COMMENTED** [review]
It looks great to me, just some thoughts I came up with

File: src/ws/app.rs [hunk 1]
```diff
@@ -91,6 +144,39 @@ pub async fn run_ws_transport(
...
+}
+
+fn to_response(event_update: Event) -> SubscribeFriendshipEventsUpdatesResponse {
+    match event_update.friendship_event {
+        Some(update) => SubscribeFriendshipEventsUpdatesResponse {
+            events: [update].to_vec(),
+        },
+        None => {
+            log::error!("There was an error when retrieving an event: Empty event");
+            panic!("There was an error when retrieving an event");
```
[the remark this answers is not in the document]
Comment (agusaldasoro) [reply src/ws/app.rs:175]: fixed!

File: src/ws/app.rs [hunk 2]
```diff
@@ -91,6 +144,39 @@ pub async fn run_ws_transport(
...
+        async move {
+            let subs_lock = subscriptions.read().await;
+            if let Some(generator) = subs_lock.get(&event_update.to.to_lowercase()) {
+                if generator.r#yield(to_response(event_update)).await.is_err() {
+                    log::error!("Event Update received > Couldn't send update to subscriptors");
+                }
+            }
+        }
+    });
+}
```
[the remark this answers is not in the document]
Comment (agusaldasoro) [reply src/ws/app.rs]: Yes, it was really confusing. I applied all your feedback 🙌

**Review by Julieta11 — APPROVED** [review]
Amazing! Just a nit comment:

I think we can simplify the imports by grouping together the repeated `friendships` module instead of importing each item individually. 

Each item individually:
```rust
use crate::{
    entities::friendship_history::FriendshipRequestEvent,
    ...
    friendships::FriendshipEventResponse,
    friendships::RejectResponse,
    friendships::RequestEvents,
    friendships::RequestResponse,
    ...
    models::friendship_event::FriendshipEvent,
    ws::service::{
        errors::{FriendshipsServiceError, FriendshipsServiceErrorResponse},
    },
};
```

Grouped:
```rust
use crate::{
    entities::friendship_history::FriendshipRequestEvent,
    friendships::{FriendshipEventResponse, 
       RejectResponse, 
       RequestEvents,  
       RequestResponse, ...}, 
       models::friendship_event::FriendshipEvent,
    ws::service::{
        errors::{FriendshipsServiceError, FriendshipsServiceErrorResponse},
    },
};
```

**Pull request merged**